File: src/main/java/fi/metatavu/metaform/server/script/ScriptController.java
package fi.metatavu.metaform.server.script;

import fi.metatavu.metaform.api.spec.model.MetaformScript;

import java.util.HashMap;
import java.util.List;

import javax.enterprise.context.ApplicationScoped;
import javax.inject.Inject;


/**
 * Controller for Metaform scripts
 */
@ApplicationScoped
public class ScriptController {

  @Inject
  private ScriptProcessor scriptProcessor;
  
  /**
   * Runs given scripts
   * 
   * @param scripts scripts
   */
  public void runScripts(List<MetaformScript> scripts) {
    if (scripts != null) {
      scripts.stream().forEach(this::runScript);
    }
  }
  
  /**
   * Runs given script
   * 
   * @param script
   */
  private void runScript(MetaformScript script) {
    if (script != null) {
      scriptProcessor.processScript(new RunnableScript(script.getLanguage(), script.getContent(), script.getName()), new HashMap<>());
    }
  }

}

File: src/main/kotlin/fi/metatavu/metaform/server/controllers/ScriptController.kt
package fi.metatavu.metaform.server.controllers

import fi.metatavu.metaform.server.persistence.model.Script
import fi.metatavu.metaform.server.script.RunnableScript
import fi.metatavu.metaform.server.script.ScriptProcessor
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject

/**
 * Controller for Metaform scripts
 */
@ApplicationScoped
class ScriptController {

    @Inject
    lateinit var scriptProcessor: ScriptProcessor

    /**
     * Runs given scripts
     *
     * @param scripts scripts
     */
    fun runScripts(scripts: List<Script>?) {
        scripts?.forEach { runScript(it) }
    }

    /**
     * Runs given script
     *
     * @param script
     */
    private fun runScript(script: Script?) {
        if (script != null) {
            scriptProcessor.processScript(RunnableScript(script.language, script.content, script.name), mapOf())
        }
    }
}